clients/privacy-center/cypress/e2e/consent-banner-gpp.cy.ts

```diff
@@ -554,4 +554,202 @@ describe("Fides-js GPP extension", () => {
+        cy.window().then((win) => {
+          win.__gpp("addEventListener", cy.stub().as("gppListener"));
+        });
+      });
+    });
+
+    it("loads the gpp extension if it is enabled", () => {
+      cy.window().then((win) => {
+        win.__gpp("ping", cy.stub().as("gppPing"));
+        cy.get("@gppPing")
+          .should("have.been.calledOnce")
+          .its("lastCall.args")
+          .then(([data, success]) => {
+            expect(success).to.eql(true);
+            expect(data.signalStatus).to.eql("ready");
+            expect(data.applicableSections).to.eql([-1]);
+          });
+      });
+    });
+
+    it("can go through the flow of user opting in to data sales and sharing", () => {
+      cy.get("button").contains("Opt in to all").click();
+      cy.waitUntilCookieExists(CONSENT_COOKIE_NAME).then(() => {
+        cy.getCookie(CONSENT_COOKIE_NAME).then((cookie) => {
+          const fidesCookie: FidesCookie = JSON.parse(
+            decodeURIComponent(cookie!.value)
+          );
+          const { consent } = fidesCookie;
+          expect(consent).to.eql({ data_sales_sharing_gpp_us_state: true });
+        });
+      });
+
+      const expected = [
+        // Empty gpp strings since user is not in GPP region
+        {
+          eventName: "listenerRegistered",
+          data: true,
+          gppString: "DBAA",
+        },
+        {
+          eventName: "cmpDisplayStatus",
+          data: "hidden",
+          gppString: "DBAA",
+        },
+        {
+          eventName: "signalStatus",
+          data: "ready",
+          gppString: "DBAA",
+          applicableSections: [-1],
+        },
+      ];
+      // Check the GPP events
+      cy.get("@gppListener")
+        .its("args")
+        .then(
+          (
+            args: [
+              { eventName: string; data: string | boolean; pingData: any },
+              boolean
+            ][]
+          ) => {
+            args.forEach(([data, success], idx) => {
+              expect(success).to.eql(true);
+              expect(data.eventName).to.eql(expected[idx].eventName);
+              expect(data.data).to.eql(expected[idx].data);
+              expect(data.pingData.gppString).to.eql(expected[idx].gppString);
+              if (expected[idx].applicableSections) {
+                expect(data.pingData.applicableSections).to.eql(
+                  expected[idx].applicableSections
+                );
+              }
+            });
+          }
+        );
+      cy.get("@gppListener")
+        .its("lastCall.args")
+        .then((args) => {
+          const [data] = args;
+          expect(data.pingData.applicableSections).to.eql([-1]);
+          // TODO: once locations and regulations are set, this value may change as it is currently hard coded
+          expect(data.pingData.supportedAPIs).to.eql([
+            "8:uscav1",
+            "10:uscov1",
+            "12:usctv1",
+            "11:usutv1",
+            "9:usvav1",
+          ]);
```

Comment:
Do we know if this still expected? It feels "wrong" but might not matter.

Reply:
This is intentional. Because `supported APIs` should include all locations configured with GPP in Admin-UI, we don't currently have a way to query for this, which is why we hard-code all possible GPP states. But we have https://ethyca.atlassian.net/browse/PROD-1595 for update the supportedAPIs specifically for state-by-state approach